src/services/catalog.service.ts

```diff
@@ -0,0 +1,152 @@
+import * as _ from 'lodash';
+
+export class CatalogService {
+  static $inject = ['Constants', '$filter'];
+
+  public $filter: any;
+  private constants: any;
+  private categories: any;
+
+  constructor(constants: any, $filter: any, $scope: any) {
+    this.constants = constants;
+    this.$filter = $filter;
+    this.categories = this.constants.SERVICE_CATALOG_CATEGORIES;
+  }
+
+  public getServiceItem(resource: any) {
+    return new ServiceItem(resource, this);
+  }
+
+  public getImageItem(resource: any) {
+    return new ImageItem(resource, this);
+  }
+
+  public getCategoriesBySubCategories(tags: any) {
+    let catsBySubCats = {};
+    _.each(tags, (tag) => {
+      _.each(this.categories, (category) => {
+        let subCat: any = _.find(category.subCategories, (subCategory: any) => {
+              return subCategory.id === tag || _.includes(subCategory.categoryAliases, tag);
+            }
+        );
+        if (subCat) {
+          catsBySubCats[subCat.id] = category.id;
+          return false;
+        }
+      });  // .ea category
+    });
+    return catsBySubCats;
+  }
+}
+
+interface IServiceItem {
+  iconClass: string;
+  imageUrl: string;
+  name: string;
+  catsBySubCats: any;
+  description: string;
+  longDescription: string;
+  resource: any;
+}
+
+export class ServiceItem implements IServiceItem {
+  public iconClass: string;
+  public imageUrl: string;
+  public name: string;
+  public catsBySubCats: any;
+  public description: string;
+  public longDescription: string;
+  public resource: any;
+  private catalogSrv: CatalogService;
+
+  constructor (serviceClass: any, catalogSrv: CatalogService) {
+    this.resource = serviceClass;
+    this.catalogSrv = catalogSrv;
+    this.imageUrl = this.getImage();
+    this.iconClass = this.getIcon();
+    this.name = this.getName();
+    this.description = this.getDescription();
+    this.longDescription = this.getLongDescription();
+    this.catsBySubCats = this.getCategoriesBySubCategories();
+  }
+
+  private getImage() {
+    return _.get(this.resource, 'metadata.imageUrl', '');
+  }
+
+  private getIcon() {
+    return 'fa fa-cubes';
```

Comment:
so based on the discussion with API folks, i think this becomes
```
return _.get(this.resource, ['osbMetadata', 'console.openshift.io/iconClass'], 'fa fa-cubes');
```
And we can debate later if there is a better default icon for service class items vs what we use for builder images.